rep-0111.rst

```diff
@@ -264,13 +264,79 @@ For backwards compatibility, the ``PACKAGE_MANAGER`` is allowed to be
 optional in the ROS Electric case.  As both ``PACKAGE_MANAGER`` and
 ``OS_VERSION`` are optional, this creates an ambiguious case where
 either ``OS_VERSION`` or ``PACKAGE_MANAGER`` is specified, but not
-both.  
+both.
 
 In this ambiguous case, rosdep first interprets the key as a
 ``PACKAGE_MANAGER``.  If this test fails, it will be interpreted as an
 ``OS_VERSION``.  Developers should exercise caution in keeping
 ``OS_VERSION`` and ``PACKAGE_MANAGER`` keys globally distinct.
 
+
+Wildcard OS_VERSION
+-------------------
+
+Many rosdep keys exist with a generic mapping independent of the OS version.
+This is great since the rule doesn't have to be updated for new OS versions as
+long as the system package name stays the same.
+But when at some point the system package name does change there are only two
+choices which are both not optimal:
+
+* Either all OS versions need to be explicitly enumerated and from that point
+  on forward the entry needs to be updated for every new OS version or
+* at some point the "old" OS versions before the system package was renamed are
+  being dropped to collapse the mapping back into a generic rule again (in
+  order to not require to update the entry in the future).
+
+To avoid either of these downsides the OS_VERSION can be specified as `*`.
+When a specific rule is being looked up and the desired OS version isn't in the
+dictionary but an OS_VERSION `*` is present it will be used as a fallback
+(instead of failing the lookup).
+
+This allows "future proofing" rosdep rules and avoids the need to update them
+for every future OS version without requiring an order on those to be
+defined (which would need to be maintained too).
+
+
+Allow PACKAGE_ARGUMENT to be explicitly null
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since the wildcard matches every not explicitly defined OS_VERSION it is not
+usable as-is in the following example:
+
+::
+
+    rosdep_name:
+      debian:
+        wheezy: [some-name]
+        jessie: [other-name]
```

Comment:
May not be essential to this PR, but is alphabetical order requirement for entries not holding here?

Reply:
Yes, it does. Updated.